flight/Modules/GPS/GPS.c

```diff
@@ -261,20 +260,22 @@ static void gpsTask(void *parameters)
 			}
 
 			if (res == PARSER_COMPLETE) {
-				timeNowMs = PIOS_Thread_Systime();
-				timeOfLastUpdateMs = timeNowMs;
-				timeOfLastCommandMs = timeNowMs;
+				timeOfLastUpdateMs = PIOS_Thread_Systime();
 			}
 		}
 
 		// Check for GPS timeout
-		timeNowMs = PIOS_Thread_Systime();
+		uint32_t timeNowMs = PIOS_Thread_Systime();
 		if ((timeNowMs - timeOfLastUpdateMs) >= GPS_TIMEOUT_MS) {
 			// we have not received any valid GPS sentences for a while.
 			// either the GPS is not plugged in or a hardware problem or the GPS has locked up.
 			uint8_t status = GPSPOSITION_STATUS_NOGPS;
 			GPSPositionStatusSet(&status);
 			AlarmsSet(SYSTEMALARMS_ALARM_GPS, SYSTEMALARMS_ALARM_ERROR);
+			/* Don't reinitialize too often. */
+			if ((timeNowMs - timeOfLastCommandMs) >= GPS_TIMEOUT_MS) {
```

Comment:
Ah, I see what this variable```timeOfLastCommandMs``` does now. Could you actually call it something like timeOfLastReinitMs ?